plugins/inventory/zabbix_inventory.py

```diff
@@ -0,0 +1,380 @@
+    - constructed
+    - inventory_cache
+'''
+
+EXAMPLES = r'''
+# Simple Inventory Plugin example
+# This will create an inventory with details from zabbix such as applications name, applicaitonids, Parent Template Name, and group membership name
+#It will also create 2 ansible inventory groups for enabled and disabled hosts in zabbix based on the status field.
+plugin: community.zabbix.zabbix_inventory
+server_url: https://zabbix.com
+login_user: Admin
+login_password: password
+host_zapi_query:
+  selectApplications: ['name', 'applicationid']
+  selectParentTemplates: ['name']
+  selectGroups: ['name']
+validate_certs: no
+groups:
+  enabled: zbx_status == "0"
+  disabled: zbx_status == "1"
+
+
+#Using Keyed Groups
+plugin: community.zabbix.zabbix_inventory
+server_url: https://zabbix.com
+login_user: Admin
+login_password: password
+validate_certs: no
+keyed_groups:
+  - key: zbx_status | lower
+    prefix: 'env'
+  - key: zbx_description | lower
+    prefix: 'test'
+    separator: ''
+
+#Using proxy format of proxy is 'http://<user>:<pass>@<proxy>:<port>' or 'http://<proxy>:<port>'
+plugin: community.zabbix.zabbix_inventory
+server_url: https://zabbix.com
+proxy: http://someproxy:8080
+login_user: Admin
+login_password: password
+validate_certs: no
+
+#Organize inventory groups based on zabbix host groups
+plugin: community.zabbix.zabbix_inventory
+server_url: https://zabbix.com
+add_zabbix_groups: True
+login_user: Admin
+login_password: password
+validate_certs: no
+
+#Using compose to modify vars
+plugin: community.zabbix.zabbix_inventory
+server_url: https://zabbix.com
+login_user: Admin
+login_password: password
+validate_certs: no
+compose:
+  zbx_testvar: zbx_status.replace("1", "Disabled")
+
+
+'''
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable, to_safe_group_name
+import os
+import atexit
+import traceback
+
+try:
+    from zabbix_api import ZabbixAPI
+    HAS_ZABBIX_API = True
+except ImportError:
+    ZBX_IMP_ERR = traceback.format_exc()
+    HAS_ZABBIX_API = False
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'community.zabbix.zabbix_inventory'
+
+    def login_zabbix(self):
+        # set proxy information if required
+        proxy = self.get_option('proxy')
+        os.environ['http_proxy'] = proxy
+        os.environ['HTTP_PROXY'] = proxy
+        os.environ['https_proxy'] = proxy
+        os.environ['HTTPS_PROXY'] = proxy
+
+        server_url = self.get_option('server_url')
+        http_login_user = self.get_option('login_user')
+        http_login_password = self.get_option('login_password')
+        validate_certs = self.get_option('validate_certs')
+        timeout = self.get_option('timeout')
+        self._zapi = ZabbixAPI(server_url, timeout=timeout, user=http_login_user, passwd=http_login_password, validate_certs=validate_certs)
+        self.login()
+        self._zbx_api_version = self._zapi.api_version()[:5]
```

Comment:
Can you please also add `logout` handling? See this [line of code](https://github.com/ansible-collections/community.zabbix/blob/1fc3d457949a9e1f7857cdea94e462cb728fcd92/plugins/module_utils/wrappers.py#L56)

Maybe you would even be able to utilize whole wrapper?

Reply:
I tried to use existing wrapper but ran into a few issues where it was prompting for a module name, so for sake of time, I added in the login piece to the inventory plugin and atexit.register(self._zapi.logout) for the logout. I believe I have added in the required documentation I am unsure how to generate the ansible community docs as I new to contributing but I updated the readme to point to the zabbix inventory plugin. Please let me know if this looks correct

Reply:
fair enough :)